### PR TITLE
Fix resteasy offsetdatetime provier

### PR DESCRIPTION
**Pull request by chameleon82**

### CODE REVIEW

**Comment by wing328** [comment]
@chameleon82 thanks for the PR. Please run the following to update the Petstore sample and commit the change so that the CI server can test the change:
```
./bin/jaxrs-resteasy-joda-petstore-server.sh
./bin/jaxrs-resteasy-petstore-server.sh
```

**Comment by wing328** [comment]
There are some errors reported by the CI. Here is part of the error messages:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project swagger-jaxrs-resteasy-server: Compilation failure: Compilation failure:
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[7,29] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[18,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[19,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[29,23] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApiService.java:[6,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[181,35] error: cannot find symbol
[ERROR] symbol:   class FormDataContentDisposition
[ERROR] location: class PetApi
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApiService.java:[29,106] error: cannot find symbol
[ERROR] symbol:   class FormDataContentDisposition
[ERROR] location: class PetApiService
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/UserApiService.java:[6,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/StoreApiService.java:[6,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/StoreApi.java:[7,29] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/StoreApi.java:[17,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/StoreApi.java:[18,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/StoreApi.java:[28,23] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/UserApi.java:[7,29] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/UserApi.java:[17,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/UserApi.java:[18,43] error: package org.glassfish.jersey.media.multipart does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/UserApi.java:[28,23] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[46,28] error: cannot find symbol
[ERROR] symbol:   class ApiParam
[ERROR] location: class PetApi
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[38,27] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[44,27] error: package io.swagger.annotations does not exist
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java:[63,31] error: cannot find symbol
[ERROR] symbol:   class ApiParam
```
Full log: https://s3.amazonaws.com/archive.travis-ci.org/jobs/146542250/log.txt

Please review and let us know if you need help fixing it.

File: samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java [hunk 1]
```diff
@@ -4,90 +4,183 @@
...
 import io.swagger.model.ModelApiResponse;
+import java.io.File;
 
 import java.util.List;
 import io.swagger.api.NotFoundException;
 
 import java.io.InputStream;
 
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
```
Comment (cbornet) [inline samples/server/petstore/jaxrs-resteasy/default/src/gen/java/io/swagger/api/PetApi.java]: It seems the sample generation went wrong...

**Comment by chameleon82** [comment]
Hm, looks like no problems in resteasy part

**Comment by chameleon82** [comment]
Hello @wing328, PR is finished and can be merged.

**Comment by wing328** [comment]
@chameleon82 thanks. I'll take another look.

Can you tell us a little bit more about the fix?

**Comment by chameleon82** [comment]
@wing328 . First, im fixed typo in offsetDateType variable (it was been the same as type and produce collision) and remove unnecessary annotations in provider classes.
Second, i try to generate samples and found typo in template paths. Next, im successfully generate samples tests. Its fall down cause artifact names was been identical ( swagger-jaxrs-resteasy-server ) in both projects. Also, fixed it. Thats all.

**Comment by wing328** [comment]
@chameleon82 thanks for the details. I did a search for `offsetDateType` in the "File changed" tab but couldn't find anything. Is the fix not committed somehow?

**Comment by chameleon82** [comment]
@wing328 here (see early eff4f3a) https://github.com/swagger-api/swagger-codegen/pull/3332/commits/eff4f3aa5a75baff11589c118f8e054f1cfdd121

**Comment by chameleon82** [comment]
@wing328 if this PR is not good i can rework it in new PR

**Comment by wing328** [comment]
@chameleon82 thanks for the pointer. It looks good. I'll merge shortly after doing some tests.

Thanks again for the PR.

**Comment by wing328** [comment]
@chameleon82 PR merged into master. Thanks again for your contribution.

**Pull request merged**